Content.Server/Power/EntitySystems/PowerMonitoringConsoleSystem.EventHandling.cs

```diff
@@ -0,0 +1,183 @@
+using Content.Server.GameTicking.Rules.Components;
+using Content.Server.NodeContainer.EntitySystems;
+using Content.Server.Power.Components;
+using Content.Server.StationEvents.Components;
+using Content.Shared.Pinpointer;
+using Content.Shared.Power;
+using Robust.Server.GameStates;
+using Robust.Shared.Map.Components;
+
+namespace Content.Server.Power.EntitySystems;
+
+internal sealed partial class PowerMonitoringConsoleSystem
+{
+    private void OnComponentStartup(EntityUid uid, PowerMonitoringConsoleComponent component, ComponentStartup args)
+    {
+        var xform = Transform(uid);
+        if (xform.GridUid == null)
+            return;
+
+        RefreshGrid(uid, component, xform.GridUid.Value, Comp<MapGridComponent>(xform.GridUid.Value));
+    }
```

Comment:
Since all consoles on the same grid will have the same chunk data, the chunk data is now updated on a per grid basis instead of on a per component basis (stored in a dictionary under PowerMonitoringConsoleSystem)